src/core/ext/load_reporting/load_reporting.h

```diff
@@ -34,19 +34,51 @@
 #ifndef GRPC_CORE_EXT_LOAD_REPORTING_LOAD_REPORTING_H
 #define GRPC_CORE_EXT_LOAD_REPORTING_LOAD_REPORTING_H
 
-#include "src/core/lib/iomgr/closure.h"
-#include "src/core/lib/surface/call.h"
+#include <grpc/impl/codegen/grpc_types.h>
+#include "src/core/lib/channel/channel_stack.h"
+
+/** Metadata key for initial metadata coming from clients */
+#define GRPC_LOAD_REPORTING_INITIAL_MD_KEY "load-reporting-initial"
+
+/** Metadata key for trailing metadata from servers */
+#define GRPC_LOAD_REPORTING_TRAILING_MD_KEY "load-reporting-trailing"
 
 typedef struct grpc_load_reporting_config grpc_load_reporting_config;
 
-/** Call information to be passed to the provided load reporting function upon
- * completion of the call */
+/** Identifiers for the invocation point of the users LR callback */
+typedef enum grpc_load_reporting_source {
+  GRPC_LR_POINT_UNKNOWN = 0,
+  GRPC_LR_POINT_CHANNEL_CREATION,
+  GRPC_LR_POINT_CHANNEL_DESTRUCTION,
+  GRPC_LR_POINT_CALL_CREATION,
+  GRPC_LR_POINT_CALL_DESTRUCTION
+} grpc_load_reporting_source;
+
+/** Call information to be passed to the provided LR callback. */
 typedef struct grpc_load_reporting_call_data {
-  const grpc_call_stats *stats;   /**< Stats for the call */
-  const char *trailing_md_string; /**< LR trailing metadata info */
+  const grpc_load_reporting_source source; /**< point of last data update. */
+
+  // XXX
```

Comment:
it's a mark I put to remember to do something, in this case add more comments. This PR is still not ready for review (but almost), that's why it doesn't have anyone assigned to it :)